st2common/st2common/triggers.py

```diff
@@ -32,75 +33,85 @@
 
 LOG = logging.getLogger(__name__)
 
-ACTION_SENSOR_ENABLED = cfg.CONF.action_sensor.enable
-TRIGGER_TYPE_ENDPOINT = cfg.CONF.action_sensor.triggers_base_url
-HTTP_POST_HEADER = {'content-type': 'application/json'}
-RETRY_WAIT = cfg.CONF.action_sensor.retry_wait
-TIMEOUT = cfg.CONF.action_sensor.request_timeout
-MAX_ATTEMPTS = cfg.CONF.action_sensor.max_attempts
 
+class InternalTriggerTypesRegistrar(object):
+
+    def __init__(self):
+        self._action_sensor_enabled = cfg.CONF.action_sensor.enable
+        self._trigger_type_endpoint = cfg.CONF.action_sensor.triggers_base_url
+        self._retry_wait = cfg.CONF.action_sensor.retry_wait
+        self._timeout = cfg.CONF.action_sensor.request_timeout
+        self._max_attempts = cfg.CONF.action_sensor.max_attempts
+        self._auth_creds = create_token('system.internal_trigger_registrar',
+                                        ttl=(1 * 60 * 60))
+        self._http_post_headers = {'content-type': 'application/json',
+                                   'X-Auth-Token': self._auth_creds.token}
+        self._get_headers = {'X-Auth-Token': self._auth_creds.token}
+
+    def register_internal_trigger_types(self):
+        LOG.debug('Registering internal trigger types...')
+
+        for resource_name, trigger_definitions in INTERNAL_TRIGGER_TYPES.items():
+            for trigger_definition in trigger_definitions:
+                LOG.debug('Registering internal trigger: %s', trigger_definition['name'])
+                if (trigger_definition['name'] == ACTION_SENSOR_TRIGGER['name'] and
+                        not self._action_sensor_enabled):
+                    continue
+                self._register_trigger_type(trigger_definition=trigger_definition, attempt_no=0)
+
+        delete_token(self._auth_creds.token)
```

Comment:
__del__, well, it doesn't really work as a destructor in Python and it's a bad idea to use it :P